### PR TITLE
Added support for pbs-torque clusters

### PR DESCRIPTION
**Pull request by matthdsm**
pull request created in follow up of #17 
I've used this edited role to deploy our local galaxy instance and I'm not having any issues.

It's possible however there needs to be some more work done to make it applicable for a wider audience.

Greetings
Matthias

### CODE REVIEW

File: tasks/pbs.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  when: galaxy_extras_install_packages
+
+- name: "Fetch DRMAA egg for Galaxy"
+  shell: "{{ galaxy_venv_dir }}/bin/python {{ galaxy_root }}/scripts/fetch_eggs.py -e drmaa -c {{ galaxy_config_file }}"
+
+- name: "Install Galaxy job conf"
+  template: src=job_conf.xml.j2 dest={{galaxy_job_conf_path}}
+
+- name: "Set PBS/torque server name"
+  shell: "echo {{ pbs_server_name }} >> /etc/torque/server_name"
```
Comment (mvdbeek) [inline tasks/pbs.yml]: this could be [lineinfile](https://docs.ansible.com/ansible/lineinfile_module.html), right?

**Comment by mvdbeek** [comment]
@matthdsm Great, this looks really nice, thank you! The only thing I'm worried about is `become` ... I'm not sure we want break compatibility with ansible-1.X already, given that 2.0 is out just a few days ago and it does support the old `sudo`/`sudo_user` combination. Also you'll have to merge with the current master.
Any other thoughts @bgruening @natefoo ?

File: defaults/main.yml [hunk 2]
```diff
@@ -2,8 +2,11 @@
 # https://github.com/bgruening/docker-galaxy-stable.
 
 galaxy_extras_config_nginx: true
+galaxy_extras_config_nginx_upload: true
 galaxy_extras_config_proftpd: true
 galaxy_extras_config_slurm: true
+galaxy_extras_config_pbs: false
+pbs_server_name: pbsqueue
```
Comment (mvdbeek) [inline defaults/main.yml]: `pbs_server_name` corresponds to the name/fqdn/ip of PBS-cluster head node?
could you add a comment here?

File: tasks/pbs.yml (hunk 1 above)
Comment (matthdsm) [inline tasks/pbs.yml]: I'm not really sure. I think I used this formulation because the "server_name" file doesn't get created when you install the torque package. 
lineinfile can't create any files or can it?

Comment (mvdbeek) [inline tasks/pbs.yml]: `state=present create=yes` should work according to the doc ...

Comment (matthdsm) [inline tasks/pbs.yml]: you're right, I'll start editing the code when you're done going through it ;)

Comment (mvdbeek) [inline tasks/pbs.yml]: I think I'm OK, but I can't guarantee you'll not have comments by other people ;)

Comment (matthdsm) [inline tasks/pbs.yml]: I'll also wait a day, so our friends on the other side of the globe get a chance to review it.

File: tasks/pbs.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+- name: Install pip
```
Comment (bgruening) [inline tasks/pbs.yml]: Why is this needed. You should be in a virutalenv and this should have pip installed, isn't it?

Comment (matthdsm) [inline tasks/pbs.yml]: True! I'll remove it when I start editing

Comment (natefoo) [inline tasks/pbs.yml]: @jmchilton wrote slurm.yml. It appears to be installed there so that we can subsequently `pip install psutil`, which isn't needed for Torque, so this can go.

File: tasks/pbs.yml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+- name: Install PBS/torque system packages
+  apt: pkg={{ item }} state={{ galaxy_extras_apt_package_state }}
+  with_items:
+    - torque-client
+    - pbs-drmaa-dev
+    - python-psutil
+  when: galaxy_extras_install_packages
+
+- name: "Fetch DRMAA egg for Galaxy"
```
Comment (natefoo) [inline tasks/pbs.yml]: Galaxy is switching from eggs to wheels (16.01 will be the first release using wheels), see https://github.com/galaxyproject/ansible-galaxy-extras/blob/master/tasks/slurm.yml#L46 for how this is handled in the slurm tasks.

Comment (matthdsm) [inline tasks/pbs.yml]: Hi nate, thanks for chiming in.
What will be the implications of the move to python wheels in 16.01? I don't think there will be any issues between slurm and torque, since in the current configuration torque also uses DRMAA.

M

Comment (natefoo) [inline tasks/pbs.yml]: Right, you should only need to incorporate the recent changes from `slurm.yml` that handle the move to wheels.

**Comment by matthdsm** [comment]
I think all issues should be resolved. If any more should pop up, let me know! Otherwise, merge away!

M

**Comment by bgruening** [comment]
@matthdsm can you please rebase this PR. Thanks :)!!!

**Comment by matthdsm** [comment]
Done! The tests seem to be running.

Matthias

Op 20 jan. 2016, om 15:52 heeft Björn Grüning <notifications@github.com<mailto:notifications@github.com>> het volgende geschreven:


@matthdsm<https://github.com/matthdsm> can you please rebase this PR. Thanks :)!!!

File: tasks/galaxy_root.yml [hunk 5]
```diff
@@ -1,10 +1,16 @@
 # TODO: following task is kind of a crutch - shouldn't be needed.
 - name: "Run common_startup.sh to create welcome.html from sample"
   shell: "cd {{ galaxy_root}} && scripts/common_startup.sh"
-  sudo: True
-  sudo_user: "{{ galaxy_user_name }}"
+  become: yes
+  become_method: sudo
 
 - name: "Install watchdog for galaxy"
+<<<<<<< HEAD
```
Comment (bgruening) [inline tasks/galaxy_root.yml]: There is a merge conflict here ...

File: tasks/uwsgi.yml [hunk 6]
```diff
@@ -1,4 +1,10 @@
 - name: "Install uwsgi for galaxy"
+<<<<<<< HEAD
+  shell: ". {{ galaxy_venv_dir }}/bin/activate && pip install uwsgi"
```
Comment (bgruening) [inline tasks/uwsgi.yml]: Here as well ...

File: templates/nginx.conf.j2 [hunk 7]
```diff
@@ -129,11 +129,15 @@ http {
             alias {{ galaxy_root }}/config/plugins/visualizations/$vis_name/static/$static_file;
         }
 
+<<<<<<< HEAD
```
Comment (bgruening) [inline templates/nginx.conf.j2]: And here ...

**Comment by matthdsm** [comment]
Sorry, I seem to have been too hasty with pushing the code.
I resolved the merge conflicts.

Matthias

**Comment by natefoo** [comment]
@matthdsm could you do the reversion from `become` and `become_user` to `sudo` and `sudo_user` as explained by @mvdbeek please?

File: templates/job_conf.xml.j2 [hunk 8]
```diff
@@ -1,36 +1,53 @@
...
+  {% endfor %}
 {% endif %}
     </handlers>
     <destinations default="cluster">
+{% if galaxy_extras_config_slurm == True %}
         <destination id="cluster" runner="slurm">
-{% if galaxy_source_shellrc %}
-            <env file="{{ galaxy_user_shellrc }}" />
+  {% if galaxy_source_shellrc %}
+          <env file="{{ galaxy_user_shellrc }}" />
```
Comment (mvdbeek) [inline templates/job_conf.xml.j2]: @matthdsm is this duplication necessary?
Is this to be able to have both slurm and pbs configured at the same time?

Comment (matthdsm) [inline templates/job_conf.xml.j2]: Good eye, none of the duplicated code was really needed. I removed it and added a conditional instead.

**Comment by mvdbeek** [comment]
Okay, thanks a lot @matthdsm . If noone objects I will merge this tomorrow.

**Comment by matthdsm** [comment]
Huraah, I’ll be my first contribution to something open source!
I’m a big boy now! :)

**Comment by mvdbeek** [comment]
This calls for a :beer:   :laughing: !

**Comment by martenson** [comment]
Thank you very much @matthdsm !

**Comment by bgruening** [comment]
Oh this is awesome! Looking forward to offer this in the Docker Image :)
Thanks a bunch @matthdsm! :beers: !!!
